src/calendar/view/CalendarAgendaView.ts

```diff
@@ -28,38 +32,92 @@ type Attrs = {
 	onEventClicked: CalendarEventBubbleClickHandler
 	groupColors: GroupColors
 	hiddenCalendars: ReadonlySet<Id>
+	startOfTheWeekOffset: number
+	isDaySelectorExpanded: boolean
 	onDateSelected: (date: Date) => unknown
 }
 
 export class CalendarAgendaView implements Component<Attrs> {
-	view({ attrs }: Vnode<Attrs>): Children {
-		const now = new Date()
-		const zone = getTimeZone()
-		const today = getStartOfDayWithZone(now, zone)
-		const tomorrow = incrementDate(new Date(today), 1)
-		const days = getNextFourteenDays(today)
-		const lastDay = lastThrow(days)
+	private selectedDate = DateTime.fromJSDate(new Date()).set({ hour: 0, minute: 0, second: 0, millisecond: 0 }).toJSDate()
+	private zone = getTimeZone()
+	private today = getStartOfDayWithZone(this.selectedDate, this.zone)
+	private tomorrow = incrementDate(new Date(this.today), 1)
+	private days = getNextFourteenDays(this.today)
+	private lastDay = lastThrow(this.days)
+	private lastDayFormatted = formatDate(this.lastDay)
+	private containerDom: HTMLElement | null = null
+	private swipeHandler!: CalendarSwipeHandler
+	private isDaySelectorExpanded = false
+	private readonly onDateSelected: (date: Date) => unknown
+
+	constructor({ attrs }: Vnode<Attrs>) {
+		this.onDateSelected = attrs.onDateSelected
+	}
+
+	private handleDayChange = (selectedDate: Date) => {
+		this.selectedDate = selectedDate
+		this.today = getStartOfDayWithZone(this.selectedDate, this.zone)
+		this.days = getNextFourteenDays(this.today)
+
+		this.onDateSelected(selectedDate)
+	}
+
+	private handleCalendarSwipe = (isNext: boolean) => {
```

Comment:
This is for the mini day-selector calendar right? There is another function named exactly the same for "main" calendar, we may want to name this differently to differentiate.